kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/modem/quectel/bg96/QuectelBG96.java

```diff
@@ -0,0 +1,375 @@
+/*******************************************************************************
+ * Copyright (c) 2011, 2016 Eurotech and/or its affiliates
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.net.admin.modem.quectel.bg96;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.kura.KuraErrorCode;
+import org.eclipse.kura.KuraException;
+import org.eclipse.kura.comm.CommConnection;
+import org.eclipse.kura.linux.net.modem.SupportedUsbModemInfo;
+import org.eclipse.kura.linux.net.modem.SupportedUsbModemsInfo;
+import org.eclipse.kura.net.admin.modem.HspaCellularModem;
+import org.eclipse.kura.net.admin.modem.hspa.HspaModem;
+import org.eclipse.kura.net.modem.ModemDevice;
+import org.eclipse.kura.net.modem.ModemPdpContext;
+import org.eclipse.kura.net.modem.ModemRegistrationStatus;
+import org.eclipse.kura.net.modem.ModemTechnologyType;
+import org.eclipse.kura.usb.UsbModemDevice;
+import org.osgi.service.io.ConnectionFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Defines Quectel BG96 modem
+ */
+public class QuectelBG96 extends HspaModem implements HspaCellularModem {
+
+    private static final Logger s_logger = LoggerFactory.getLogger(QuectelBG96.class);
+
+    private final int m_pdpContext = 1;
+
+    /**
+     * TelitHe910 modem constructor
+     *
+     * @param usbDevice
+     *            - modem USB device as {@link UsbModemDevice}
+     * @param platform
+     *            - hardware platform as {@link String}
+     * @param connectionFactory
+     *            - connection factory {@link ConnectionFactory}
+     */
+    public QuectelBG96(ModemDevice device, String platform, ConnectionFactory connectionFactory) {
+
+        super(device, platform, connectionFactory);
+
+        try {
+            String atPort = getAtPort();
+            String gpsPort = getGpsPort();
+            if (atPort != null) {
+                if (atPort.equals(getDataPort()) || atPort.equals(gpsPort)) {
```

Comment:
this if can be merged with the previous one since there are no other branches.

Reply:
Don't fully understand the comment

Reply:
You can merge the two ifs into a single one. Logically should be the same behaviour.